File: src/main/java/com/thesilentnights/easylogin/service/ActionCheckService.java
package com.thesilentnights.easylogin.service;

import com.thesilentnights.easylogin.repo.PlayerCache;
import net.minecraft.server.level.ServerPlayer;
import net.minecraft.world.entity.LivingEntity;

import java.util.UUID;

public class ActionCheckService {

    public static boolean shouldCancelEvent(LivingEntity entity) {
        if (entity instanceof ServerPlayer) {
            return !isLoggedIn(entity.getUUID()) && !ByPassService.isBypassed(entity.getUUID());
        }
        return false;
    }

    public static boolean isLoggedIn(UUID uuid) {
        return PlayerCache.hasAccount(uuid);
    }
}

File: src/main/kotlin/com/thesilentnights/easylogin/service/ActionCheckService.kt
package com.thesilentnights.easylogin.service

import com.thesilentnights.easylogin.repo.PlayerCache.hasAccount
import net.minecraft.server.level.ServerPlayer
import net.minecraft.world.entity.LivingEntity
import java.util.*

object ActionCheckService {
    @JvmStatic
    fun shouldCancelEvent(entity: LivingEntity): Boolean {
        if (entity is ServerPlayer) {
            return !isLoggedIn(entity.getUUID()) && !ByPassService.isBypassed(entity.uuid)
        }
        return false
    }

    fun isLoggedIn(uuid: UUID): Boolean {
        return hasAccount(uuid)
    }
}
